### DISCUSSION TITLE
Invalid `prisma.list.update()` invocation while updating many-to-many relations

### DISCUSSION THREAD

**Discussion by faisalsayed10** [prisma/prisma#8377] [Q&A] [answered]
Hey, I'm trying to update a many-to-many relation between `Book` and `List`, so a person can add books to the list anytime. Here's how I'm doing it currently: 
```ts
    const updatedList = await prisma.list.update({
      where: { id },
      data: {
        books: {
          connectOrCreate: {
            where: { gbookId: book.id },
            create: {
              gbookId: book.id,
              title: book.title,
              authors: book.authors,
              publishedDate: book.publishedDate,
              previewLink: book.previewLink,
              imageLinks: Object.values(book.imageLinks),
            },
          },
        },
      },
      include: { books: true },
    });

    return res.status(200).json(updatedList);
```

But doing this throws an error in the console and I'm not sure how to fix it.

```
PrismaClientValidationError:
Invalid `prisma.list.update()` invocation:

{
  where: {
    id: 'd43fbfa6-2516-444c-ade6-1be2a43921d9'
  },
  data: {
    books: {
    ~~~~~
      connectOrCreate: {
        where: {
          gbookId: 'this is google books id'
        },
        create: {
          gbookId: 'this is google books id',
          title: 'this is a book title',
          authors: [
            'book.authors',
            'second author'
          ],
          publishedDate: 'book.publishedDate 1998',
          previewLink: 'https://book.previewLink',
          imageLinks: [
            '1234',
            44
          ]
        }
      }
    }
  },
  include: {
    books: true
  }
}

Unknown arg `books` in data.books for type ListUncheckedUpdateInput. Did you mean `id`? Available args:
type ListUncheckedUpdateInput {
  id?: String | StringFieldUpdateOperationsInput
  createdAt?: DateTime | DateTimeFieldUpdateOperationsInput
  updatedAt?: DateTime | DateTimeFieldUpdateOperationsInput
  title?: String | StringFieldUpdateOperationsInput
  type?: ListType | EnumListTypeFieldUpdateOperationsInput
  description?: String | StringFieldUpdateOperationsInput
  authorId?: Int | IntFieldUpdateOperationsInput
}


    at Document.validate (/Users/fayd/Projects/libook/node_modules/@prisma/client/runtime/index.js:32626:19)
    at NewPrismaClient._executeRequest (/Users/fayd/Projects/libook/node_modules/@prisma/client/runtime/index.js:34733:17)
    at consumer (/Users/fayd/Projects/libook/node_modules/@prisma/client/runtime/index.js:34678:23)
    at /Users/fayd/Projects/libook/node_modules/@prisma/client/runtime/index.js:34680:47
    at AsyncResource.runInAsyncScope (async_hooks.js:197:9)
    at NewPrismaClient._request (/Users/fayd/Projects/libook/node_modules/@prisma/client/runtime/index.js:34680:25)
    at Object.then (/Users/fayd/Projects/libook/node_modules/@prisma/client/runtime/index.js:34788:39)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  clientVersion: '2.26.0'
}
```

Any help is appreciated

**Comment by ryands17** [comment, marked as answer]
@faisalsayed10 👋 

Can you share your `schema.prisma` as well?

**Reply by faisalsayed10** [reply]
Hey! Here's my schema file:

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model List {
  id          String   @id @default(uuid())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  title       String
  type        ListType @default(normal)
  description String   @default("")
  books       Book[]

  @@map("lists")
}

model Book {
  id                  String   @id @default(uuid())
  gbookId             String   @unique @map(name: "gbook_id")
  title               String
  authors             String[]
  publishedDate       String   @map(name: "published_date")
  previewLink         String   @map(name: "preview_link")
  imageLinks          String[] @map(name: "image_links")
  lists               List[]

  @@map("books")
}

enum ListType {
  normal
  wantToRead
  currentlyReading
  alreadyRead
}

**Reply by ryands17** [reply]
@faisalsayed10 I cannot reproduce your issue. It's working fine when I run the above query:
![image](https://user-images.githubusercontent.com/19697099/126593244-f2ae52c6-da38-44ae-b9fd-e40b6918a329.png)

Can you check if `prisma` and `@prisma/cli` are on the latest version?

**Reply by faisalsayed10** [reply]
prisma and prisma client are v2.26.0

**Reply by faisalsayed10** [reply]
okay so i found what the issue was, so basically, while making the post request, I was accidentally sending an `integer` in a `string[]` for the value of `imageLinks`

**Reply by faisalsayed10** [reply]
But prisma gave a vague error and I wasn't able to understand what the issue was, until i tried changing the request body

**Reply by ryands17** [reply]
I would suggest adding a validator for your `req.body` in this case. As for the error, you can open an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) for the misleading error so that we can look into this.

**Reply by faisalsayed10** [reply]
sure thing, thanks for the help!

**Comment by yeonlala** [comment]
hello @faisalsayed10 
Maybe can try it!

` 
const updatedList = await prisma.list.update({
   where: { id },
   data: {
     Book: { //  Book -- this is important !
       connectOrCreate: {
         where: { gbookId: book.id },
         create: {
           gbookId: book.id,
           ...
         },
       },
     },
   },
   include: { books: true },
 })
`

**Reply by faisalsayed10** [reply]
`Book` throws typescript error, `Book` (as a key) does not exist

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
